src/webgpu/util/f32_interval.ts

```diff
@@ -746,26 +746,47 @@ const AtanIntervalOp: PointToIntervalOp = {
 };
 
 /** Calculate an acceptance interval of atan(x) */
-export function atanInterval(n: number): F32Interval {
+export function atanInterval(n: number | F32Interval): F32Interval {
   return runPointOp(toF32Interval(n), AtanIntervalOp);
 }
 
 const Atan2IntervalOp: BinaryToIntervalOp = {
   impl: (y: number, x: number): F32Interval => {
-    const numULP = 4096;
+    if (x === 0) {
+      // atan2 is not defined meaningfully here
+      return F32Interval.any();
```

Comment:
DXC and cppreference both say that

x = 0   y <0 should yield  -pi/2
x = 0  y  > 0 should yield pi / 2

Reply:
I agree those specific implementations do that, but if we are supporting other implementations where atan2 ~ atan(y/x) then those others one will get an undefined value when x = 0, and that would be more permissive than pi/2 or -pi/2.

I should probably rewrite the comment to be clear that it is in the atan(y/x) case that this is undefined.